src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -59,24 +59,54 @@ public class HubEndPoint<THub, TClient> : IInvocationBinder where THub : Hub<TCl
 
         public async Task OnConnectedAsync(ConnectionContext connection)
         {
-            await ProcessNegotiate(connection);
+            var output = Channel.CreateUnbounded<byte[]>();
+            var connectionContext = new HubConnectionContext(output, connection);
+
+            await ProcessNegotiate(connectionContext);
```

Comment:
Why do we need to do this? We moved to a place where the protocol itself is all we need. Once you negotiate json or protobuf, nothing else needs to be implied.